packages/uui-copy/lib/uui-copy.element.ts

```diff
@@ -0,0 +1,175 @@
+import { defineElement } from '@umbraco-ui/uui-base/lib/registration';
+import { css, html, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+import { UUIButtonElement } from '@umbraco-ui/uui-button/lib';
+import { UUICopyEvent } from './UUICopyEvent';
+import { demandCustomElement } from '@umbraco-ui/uui-base/lib/utils';
+import { LabelMixin } from '@umbraco-ui/uui-base/lib/mixins';
+
+/**
+ * @summary A button to trigger text content to be copied to the clipboard
+ * @element uui-copy
+ * @dependency uui-button
+ * @dependency uui-icon
+ * @fires {UUICopyEvent} copying - Fires before the content is about to copied to the clipboard and can be used to transform or modify the data before its added to the clipboard
+ * @fires {UUICopyEvent} copied - Fires when the content is copied to the clipboard
+ * @slot - Use to replace the default content of 'Copy' and the copy icon
+ */
+@defineElement('uui-copy')
```

Comment:
Brought the name of the component up with the team as well. Seeing that it is functionally a button, we would prefer to rename it to something a little more descriptive, e.g. **uui-text-copy-button** to indicate that it is a _button_ that copies _text_ (you can copy all sorts of things in the browser these days and we wouldn't want to confuse it with anything else like a **Copy block** button for instance).